### ISSUE TITLE
Re-organize HL7 serializer code to split read and write

### ISSUE THREAD

**Issue by carlosfelix2** [CDCgov/prime-reportstream#1202]
The file for the HL7 serializer contains a lot of code for reading and writing and has become a bit hard to follow and large.  Split the code into separate files.  Also, refactor the code that uses the mappers into common code as the files CSVSerializer and Translator also have the same logic.  Run the refactor by @RickHawesUSDS before implementing it.

Acceptance criteria:
1. Split the code for the HL7 serializer into smaller files, perhaps for read and write.
2. Refactor the use of the mapper as  common code that can be used by HL7serializer, CSVSerializer and Translator.

**Comment by RickHawesUSDS** [comment]
I would also like to have a clear separation between standard hl7, HHS conformant hl7, and state quirks.
